nav2_graceful_controller/src/graceful_controller.cpp

```diff
@@ -195,46 +196,36 @@ geometry_msgs::msg::TwistStamped GracefulController::computeVelocityCommands(
     // Else, fall through and see if we should follow control law longer
   }
 
-  // Precompute distance to candidate poses
+  // Find a valid target pose and its trajectory
+  nav_msgs::msg::Path local_plan;
+  bool valid_target_pose_found;
+  geometry_msgs::msg::PoseStamped target_pose;
+
+  double dist_to_target;
   std::vector<double> target_distances;
   computeDistanceAlongPath(transformed_plan.poses, target_distances);
 
-  // Work back from the end of plan to find valid target pose
-  for (int i = transformed_plan.poses.size() - 1; i >= 0; --i) {
-    // Underlying control law needs a single target pose, which should:
-    //  * Be as far away as possible from the robot (for smoothness)
-    //  * But no further than the max_lookahed_ distance
-    //  * Be feasible to reach in a collision free manner
-    geometry_msgs::msg::PoseStamped target_pose = transformed_plan.poses[i];
-    double dist_to_target = target_distances[i];
-
-    // Continue if target_pose is too far away from robot
-    if (dist_to_target > params_->max_lookahead) {continue;}
-
-    if (dist_to_goal < params_->max_lookahead) {
-      if (params_->prefer_final_rotation) {
-        // Avoid instability and big sweeping turns at the end of paths by
-        // ignoring final heading
-        double yaw = std::atan2(target_pose.pose.position.y, target_pose.pose.position.x);
-        target_pose.pose.orientation = nav2_util::geometry_utils::orientationAroundZAxis(yaw);
-      }
-    } else if (dist_to_target < params_->min_lookahead) {
-      // Make sure target is far enough away to avoid instability
-      break;
+  for (int i = transformed_plan.poses.size(); i >= 0; --i) {
```

Comment:
If you make this an `unsigned int` instead of an `int`, then you don't need the `static_cast` below and nothing else needs to change

Reply:
On making it unsigned int, `i>=0` condition will always remain valid.
And I am getting a compiler error for the same.

Reply:
@SakshayMahna there are many solutions

```
for (auto it = vec.rbegin(); it != vec.rend(); ++it)
for (size_t i = vec.size(); i-- > 0;)
```

Reply:
Went with this:
```cpp
bool is_first_iteration = true;
for (int i = transformed_plan.poses.size() - 1; i >= 0; --i) {
    if (is_first_iteration) ...
    else ...
}
```